pandas/core/indexes/base.py

```diff
@@ -2393,6 +2404,13 @@ def intersection(self, other):
         taken = other.take(indexer)
         if self.name != other.name:
             taken.name = None
+        if sort:
+            try:
+                taken = taken.sort_values()
+            except TypeError as e:
+                warnings.warn("%s, sort order is undefined for "
```

Comment:
does this actually get hit? I really hate all of these sort warnings. *maybe* can you factor them out in a helper function instead?

Reply:
?

Reply:
can you use .format here